### DISCUSSION TITLE
Why is Console.log() output width limited in Docker Compose logs?

### DISCUSSION THREAD

**Discussion by keyiflerolsun** [Textualize/rich#3436] [Q&A] [answered]
Hi everyone,

![image](https://github.com/user-attachments/assets/b36557bb-0316-4671-a406-b116d17a3cda)

I'm using the Rich library in my Python project to produce pretty logs with `Console.log()`. Everything works perfectly when running the application locally. However, when I run my application in a Docker container using Docker Compose, I notice that the output width of the logs seems to be limited, and lines are wrapping prematurely, which affects the readability of the logs.

Here's a brief overview of how I'm setting up Rich:

```python
from rich import pretty, traceback
from rich.console import Console

pretty.install()
traceback.install(show_locals=False)

console = Console(log_path=False)
console.log("This is a sample log message that should ideally not wrap in the middle of the sentence.")
```

In Docker Compose, the logs appear wrapped even though my terminal width should support wider lines.

Has anyone encountered similar issues? Is there a specific setting or parameter I need to adjust to ensure the logs maintain their intended width when viewed in Docker Compose? Any help or pointers would be greatly appreciated!

Thank you!

**Comment by TomJGooding** [comment, marked as answer]
I'm still not too familar with Docker but this https://github.com/Textualize/rich/issues/1747 might help?

**Reply by keyiflerolsun** [reply]
```python
console = Console(
    log_path = False,
    _environ = {"COLUMNS": "112"}
)
```

solved, thank you..

**Comment by ayorgo** [comment]
You can do it without hardcoding the value for `COLUMNS`.
Either by:
```sh
COLUMNS=$COLUMNS docker compose up --no-log-prefix test
```
or
```sh
COLUMNS=$(tput cols) docker compose up --no-log-prefix test
```
or, if you happen to be using Makefile:
```makefile
.PHONY: test
test:
	COLUMNS=$$(tput cols) \
	docker compose up --no-log-prefix test
```
The [`--no-log-prefix`](https://docs.docker.com/reference/cli/docker/compose/logs/) option removes the container name prefix from the log output.
